libc/math/common.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Math tests common part.
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "common.h"
+
+#include <stdlib.h>
+#include <time.h>
+#include <math.h>
+#include <float.h>
+
+double test_getRandomLog(double min, double max)
+{
+	if (min == 0.0) {
+		min = DBL_MIN;
+	}
+	double a = log(min);
+	double b = log(max);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+double test_getRandomLogSubnrm()
+{
+	double a = log(5.0e-324);
+	double b = log(2.0e-308);
+	double r = (double)rand() / (double)RAND_MAX;
+
+	return exp(a + r * (b - a));
+}
+
+
+void test_setup(void)
+{
+	double num = M_PI;
+	double one = 1.0;
+	double zero = 0.0;
+
+	test_common.maxPowTwoPi = 0;
+	test_common.maxPowTwo = 0;
+	test_common.dblSignif = 0;
+
+	/* Initialize seed for test */
+	srand(time(0));
+
+	/* Initialize maxPowTwoPi value */
+	do {
+		test_common.maxPowTwoPi++;
+		num *= 2.0;
+	} while (num - M_PI != num);
```

Comment:
`num` starts as `M_PI`, and we loop until `num - M_PI != num`, so until `num != 2 * M_PI`.
So I see two problems here:
- there's a equality comparison between doubles, and it's somewhat lucky this loop end at all,
- `test_common.maxPowTwoPi = 1`, no need for calculating this in the runtime.

Or did I got something wrong?

Reply:
This loop runs until `num` will be such big number that subtracting `M_PI` will result in same number, as I remember it is `test_common.maxPowTwoPi = 53` , this value is needed to tests with periodic funcs eg. `sins(x)` where big `x` is normalized to `<0, 2 * M_PI>` , potential wrong implementation could cause infinity loop continuously subtracting `M_PI` from `x`.

Reply:
Oh, ok, I misunderstood the code indeed.